iphone/Maps/Classes/Routing/MWMRouter.mm

```diff
@@ -119,6 +129,8 @@ - (void)buildFromPoint:(MWMRoutePoint const &)start
 
 - (void)rebuildWithBestRouter:(BOOL)bestRouter
 {
+  [self.altitudeImagesData removeAllObjects];
```

Comment:
Looks like these calls should be posted on the renderAltitudeImagesQueue, otherwise at least logical data race is deteced.